pkg/attacher/connection.go

```diff
@@ -57,43 +47,40 @@ type CSIConnection interface {
 	// should retry.
 	Detach(ctx context.Context, volumeID string, nodeID string, secrets map[string]string) (detached bool, err error)
 
-	// Probe checks that the CSI driver is ready to process requests
-	Probe(singleProbeTimeout time.Duration) error
-
 	// Close the connection
 	Close() error
 }
 
-type csiConnection struct {
+type attacher struct {
 	conn         *grpc.ClientConn
 	capabilities []csi.ControllerServiceCapability
 }
 
 var (
-	_ CSIConnection = &csiConnection{}
+	_ Attacher = &attacher{}
 )
 
 // New provides a new CSI connection object.
-func New(address string) (CSIConnection, error) {
+func New(address string) (Attacher, error) {
 	conn, err := connection.Connect(address, connection.OnConnectionLoss(connection.ExitOnConnectionLoss()))
 	if err != nil {
 		return nil, err
 	}
-	return &csiConnection{
+	return &attacher{
 		conn: conn,
 	}, nil
 }
 
-func (c *csiConnection) GetDriverName(ctx context.Context) (string, error) {
-	return connection.GetDriverName(ctx, c.conn)
+func (c *attacher) GetDriverName(ctx context.Context) (string, error) {
+	return rpc.GetDriverName(ctx, c.conn)
 }
 
-func (c *csiConnection) Probe(singleProbeTimeout time.Duration) error {
-	return connection.ProbeForever(c.conn, singleProbeTimeout)
+func (c *attacher) Probe(singleProbeTimeout time.Duration) error {
+	return rpc.ProbeForever(c.conn, singleProbeTimeout)
 }
 
-func (c *csiConnection) SupportsControllerPublish(ctx context.Context) (supportsControllerPublish bool, supportsPublishReadOnly bool, err error) {
-	caps, err := connection.GetControllerCapabilities(ctx, c.conn)
+func (c *attacher) SupportsControllerPublish(ctx context.Context) (supportsControllerPublish bool, supportsPublishReadOnly bool, err error) {
```

Comment:
I don't think `SupportsControllerPublish` needs to be part of attacher, let's just move this as an unexported function in main.go?